### PR TITLE
[TBTC-89] Enable rountrip test

### PR DESCRIPTION
**Pull request by Martoon-00**
## Description

Bring the previously disabled roundtrip test back to life.

I'm not sure taken solution is sensible, but don't see a viable alternative.

## Related issue(s)

https://issues.serokell.io/issue/TBTC-89

#### Related changes (conditional)

- Tests (see [short guidelines](../blob/master/CONTRIBUTING.md#tests))
  - [x] If I added new functionality, I added tests covering it.
  - [x] If I fixed a bug, I added a regression test to prevent the bug from
        silently reappearing again.

- Documentation
  - I checked whether I should update the docs and did so if necessary:
    - [x] [README](../blob/master/README.md)
    - [x] Haddock
    - [x] [docs/](../blob/master/docs/)
  - [x] I updated [changelog](../blob/master/ChangeLog.md) unless I am sure my changes are
        really minor.

#### Stylistic guide (mandatory)

- [x] My commits comply with [the following policy](https://www.notion.so/serokell/Where-and-how-to-commit-your-work-58f8973a4b3142c8abbd2e6fd5b3a08e).
- [x] My code complies with the [style guide](../blob/master/docs/code-style.md).

### CODE REVIEW

**Comment by sras** [comment]
@gromakovsky Have updated this PR to fix the test errors.

/cc @Martoon-00

File: src/Client/IO/TezosClient.hs [hunk 1]
```diff
@@ -160,6 +162,16 @@ getAddressForContract v tcPath alias  = do
         Left err -> throwM $ TzbtcParseError (show err))
       (\stderr' -> pure $ TzbtcTezosClientError stderr')
 
+getChainId
+  :: Bool -> FilePath -> Text -> IO (Either TzbtcClientError ChainId)
+getChainId v tcPath name  = do
+  callTezosClient' v tcPath
+    ["rpc", "get", "chains/" <> (toString name) <> "/chain_id"] Nothing (\stdout' ->
```
Comment (gromakovsky) [inline src/Client/IO/TezosClient.hs:169]: Why are you doing it via `tezos-client` instead of querying node directly?

Comment (sras) [reply src/Client/IO/TezosClient.hs:169]: It appeared to be a bit easier. But yes, this is an RPC call, so better done via our RPC infra.

**Comment by sras** [comment]
### Test log after changes (8b73fd5)

```
>tzbtc-client deployTzbtcContract --redeem nettest --token-name "TZBTC test" --token-symbol "TZBTC" --user nettest -v

(Had verbose logging so only including the final output..)
..
..
...
..
Fee: 0.070 Tez
Operation hash: ooyJTw4YNKEKgATcm1iUBVHDFUKwxmBqR21UjnnCJqS7oSrk5Dz
Executing call:tezos-client wait for ooyJTw4YNKEKgATcm1iUBVHDFUKwxmBqR21UjnnCJqS7oSrk5Dz to be included
Status:ExitSuccess
Output:Operation found in block: BMSQtiTxbkePG8sU3zxk8QCYMbYA9ugVdD4X9hyn9zdjLPgiZMf (pass: 3, offset: 0)

Error:Warning:

                 This is NOT the Tezos Mainnet.

     The node you are connecting to claims to be running on the
               Tezos Alphanet DEVELOPMENT NETWORK.
          Do NOT use your fundraiser keys on this network.
          Alphanet is a testing network, with free tokens.


Operation found in block: BMSQtiTxbkePG8sU3zxk8QCYMbYA9ugVdD4X9hyn9zdjLPgiZMf (pass: 3, offset: 0)
Contract was successfully deployed. Contract address: KT1PE2dHZ57ynjWY9QhWPCmiKiBWSfmdPpWQ
Right now there is no contract aliased as 'tzbtc' in tezos-client config.
Would you like to add/replace alias 'tzbtc' with the newly deployed contract? [Y/N]
Y

> tzbtc-client transferOwnership KT1Gp1Y6rcYJXV9Yqnjtv4xQQthZFWsRj31y --user nettest
Fee: 0.019 Tez
Operation hash: oospyvXJYqPdU1eur2i7fVuaUR1FLFa2L4LQi6PkUwqx8dt6eWY
Operation found in block: BLXfxq9GvLUqufiH6Q6Ez5A2aNb55iqgPVZ6jFPcFVkE1uGBDkh (pass: 3, offset: 0)
Fee: 0.009 Tez

Operation hash: opPVkN2Ph4KN4DyEp8vTbrMvPo9ykScmwSe3bwWnisn1y1ewZRP
Operation found in block: BLLkSFqhU5GWg6cShCzCTeFRLBWr6e2JQ1YyWByk6vMsYVWB1G4 (pass: 3, offset: 0)
Multisig contract was successfully deployed. Contract address: KT1Gp1Y6rcYJXV9Yqnjtv4xQQthZFWsRj31y
Right now there is no contract aliased as 'tzbtc-multisig' in tezos-client config.
Would you like to add/replace alias 'tzbtc-multisig' with the newly deployed contract? [Y/N]
Y

>tzbtc-client transferOwnership KT1Gp1Y6rcYJXV9Yqnjtv4xQQthZFWsRj31y --user nettest
Fee: 0.019 Tez
Operation hash: oospyvXJYqPdU1eur2i7fVuaUR1FLFa2L4LQi6PkUwqx8dt6eWY
Operation found in block: BLXfxq9GvLUqufiH6Q6Ez5A2aNb55iqgPVZ6jFPcFVkE1uGBDkh (pass: 3, offset: 0)

>tzbtc-client acceptOwnership -m ../../temp/multisig-test/accept-ownership --multisig
-addr KT1Gp1Y6rcYJXV9Yqnjtv4xQQthZFWsRj31y -v

(Bug with ChainId parsing was encountered here. After fixing the bug) 

> tzbtc-client acceptOwnership -m ../../temp/multisig-test/accept-ownership --multisig
-addr KT1Gp1Y6rcYJXV9Yqnjtv4xQQthZFWsRj31y -v
Executing call:tezos-client config show
Status:ExitSuccess
Output:
Error:Warning:

                 This is NOT the Tezos Mainnet.

     The node you are connecting to claims to be running on the
               Tezos Alphanet DEVELOPMENT NETWORK.
          Do NOT use your fundraiser keys on this network.
          Alphanet is a testing network, with free tokens.

Warning: no config file at /home/sras/.tezos-client/config,
  displaying the default configuration.
{ "base_dir": "/home/sras/.tezos-client", "node_addr": "localhost",
  "node_port": 8732, "tls": false, "web_port": 8080, "confirmations": 0 }


Executing call:tezos-client show known contract tzbtc
Status:ExitSuccess
Output:KT1PE2dHZ57ynjWY9QhWPCmiKiBWSfmdPpWQ


Error:Warning:

                 This is NOT the Tezos Mainnet.

     The node you are connecting to claims to be running on the
               Tezos Alphanet DEVELOPMENT NETWORK.
          Do NOT use your fundraiser keys on this network.
          Alphanet is a testing network, with free tokens.


Executing call:tezos-client show known contract tzbtc-multisig
Status:ExitSuccess
Output:KT1Gp1Y6rcYJXV9Yqnjtv4xQQthZFWsRj31y


Error:Warning:

                 This is NOT the Tezos Mainnet.

     The node you are connecting to claims to be running on the
               Tezos Alphanet DEVELOPMENT NETWORK.
          Do NOT use your fundraiser keys on this network.
          Alphanet is a testing network, with free tokens.

Executing call:tezos-client config show
Status:ExitSuccess
Output:
Error:Warning:

                 This is NOT the Tezos Mainnet.

     The node you are connecting to claims to be running on the
               Tezos Alphanet DEVELOPMENT NETWORK.
          Do NOT use your fundraiser keys on this network.
          Alphanet is a testing network, with free tokens.

Warning: no config file at /home/sras/.tezos-client/config,
  displaying the default configuration.
{ "base_dir": "/home/sras/.tezos-client", "node_addr": "localhost",
  "node_port": 8732, "tls": false, "web_port": 8080, "confirmations": 0 }


Executing call:tezos-client show known contract tzbtc
Status:ExitSuccess
Output:KT1PE2dHZ57ynjWY9QhWPCmiKiBWSfmdPpWQ


Error:Warning:

                 This is NOT the Tezos Mainnet.

     The node you are connecting to claims to be running on the
               Tezos Alphanet DEVELOPMENT NETWORK.
          Do NOT use your fundraiser keys on this network.
          Alphanet is a testing network, with free tokens.


Executing call:tezos-client show known contract tzbtc-multisig
Status:ExitSuccess
Output:KT1Gp1Y6rcYJXV9Yqnjtv4xQQthZFWsRj31y


Error:Warning:

                 This is NOT the Tezos Mainnet.

     The node you are connecting to claims to be running on the
               Tezos Alphanet DEVELOPMENT NETWORK.
          Do NOT use your fundraiser keys on this network.
          Alphanet is a testing network, with free tokens.


Executing call:tezos-client config show
Status:ExitSuccess
Output:
Error:Warning:

                 This is NOT the Tezos Mainnet.

     The node you are connecting to claims to be running on the
               Tezos Alphanet DEVELOPMENT NETWORK.
          Do NOT use your fundraiser keys on this network.
          Alphanet is a testing network, with free tokens.

Warning: no config file at /home/sras/.tezos-client/config,
  displaying the default configuration.
{ "base_dir": "/home/sras/.tezos-client", "node_addr": "localhost",
  "node_port": 8732, "tls": false, "web_port": 8080, "confirmations": 0 }


Executing call:tezos-client show known contract tzbtc
Status:ExitSuccess
Output:KT1PE2dHZ57ynjWY9QhWPCmiKiBWSfmdPpWQ


Error:Warning:

                 This is NOT the Tezos Mainnet.

               Tezos Alphanet DEVELOPMENT NETWORK.
          Do NOT use your fundraiser keys on this network.
          Alphanet is a testing network, with free tokens.

Warning: no config file at /home/sras/.tezos-client/config,
  displaying the default configuration.
{ "base_dir": "/home/sras/.tezos-client", "node_addr": "localhost",
  "node_port": 8732, "tls": false, "web_port": 8080, "confirmations": 0 }


Executing call:tezos-client show known contract tzbtc
Status:ExitSuccess
Output:KT1PE2dHZ57ynjWY9QhWPCmiKiBWSfmdPpWQ
     The node you are connecting to claims to be running on the
               Tezos Alphanet DEVELOPMENT NETWORK.
          Do NOT use your fundraiser keys on this network.
          Alphanet is a testing network, with free tokens.


Executing call:tezos-client show known contract tzbtc-multisig
Status:ExitSuccess
Output:KT1Gp1Y6rcYJXV9Yqnjtv4xQQthZFWsRj31y


Error:Warning:

                 This is NOT the Tezos Mainnet.

     The node you are connecting to claims to be running on the
               Tezos Alphanet DEVELOPMENT NETWORK.
          Do NOT use your fundraiser keys on this network.
          Alphanet is a testing network, with f

> cat ../../temp/multisig-test/accept-ownership
{"toSign":"05070707070a000000049caecab90a00000016015a421391a82688971ff68b16585a669de46c48d5000707000005050707050805080508050805080
30b0a0000001601a09db93def4d99dcd713c8ad3d9b3061f75fdc8100","signatures":[]}


>cd ../../temp/multisig-test/
> ls
accept-ownership
> cp accept-ownership accept-ownership-alice
> cp accept-ownership accept-ownership-john
> tzbtc-client signPackage --package accept-ownership-alice --user alice
You are going to sign the following package:

Operation:
------------------
Accept ownership

Included Signatures:
--------------------
[]

Are you sure? [Y/N]
Y

>tzbtc-client signPackage --package accept-ownership-john --user john
You are going to sign the following package:

Operation:
------------------
Accept ownership

Included Signatures:
--------------------
[]

Are you sure? [Y/N]
Y

>tzbtc-client callMultisig --package accept-ownership-john --package accept-owner
ship-alice --user nettest
Fee: 0.038 Tez
Operation hash: ooSawRwar7wcny3K2Yb6DdjC9JrnzT7cEhjwZh47hSU7oD5F5yK
Operation found in block: BMSjWEkJXur3Wjmc12K3dXA1eKdx88sb9tTdhgfaod2WpDTohL8 (pass: 3, offset: 0)

```

File: src/Client/IO/TezosClient.hs (hunk 1 above)
Comment (sras) [reply src/Client/IO/TezosClient.hs]: @gromakovsky Have moved it to RPC.

**Comment by gromakovsky** [comment]
Next time please set `TEZOS_CLIENT_UNSAFE_DISABLE_DISCLAIMER=YES` to make logs smaller.

File: src/Lorentz/Contracts/TZBTC/V0.hs [hunk 2]
```diff
@@ -232,8 +233,7 @@ tzbtcDoc = buildLorentzDoc $ do
     , "DX-License-Identifier: LicenseRef-MIT-BitcoinSuisse"
     ]
   contractName "TZBTC" $ do
-    doc $ $mkDGitRevision $ GitRepoSettings $
-      mappend "https://github.com/tz-wrapped/tezos-btc/commit/"
+    doc $ DGitRevisionUnknown
```
Comment (gromakovsky) [inline src/Lorentz/Contracts/TZBTC/V0.hs:236]: It should be `contractGeneralDefault`. And the place where we print contract documentation should be updated to do `buildLorentzDocWithGitRev`.

**Comment by sras** [comment]
>Don't forget to update changelog.

@gromakovsky Should I only mention the switch to the updated multisig contract?

**Comment by gromakovsky** [comment]
Yes, but this is an important change because it breaks interaction between previously deployed multisig contracts and new `tzbtc-client` (or vice versa). I suppose we should update the version to 0.5.0.

File: src/Lorentz/Contracts/TZBTC/V0.hs (hunk 2 above)
Comment (sras) [reply src/Lorentz/Contracts/TZBTC/V0.hs]: @gromakovsky I have made this change, please check.

**Comment by sras** [comment]
@gromakovsky Have updated the version to 0.5.0, and update changelog to mention this change as part of 0.5.0

File: package.yaml [hunk 3]
```diff
@@ -6,7 +6,7 @@
 
 name:                tzbtc
 # If you update this version, make sure to update it in release.nix
```
Comment (gromakovsky) [inline package.yaml:8]: You forgot :peka:

File: ChangeLog.md [hunk 4]
```diff
@@ -11,6 +11,12 @@ Changes that affect Michelson code are tagged with `[CONTRACT]`.
 Unreleased
 ==========
 
+0.5.0
+=====
+* [#112](https://github.com/tz-wrapped/tezos-btc/pull/112)
+  - Update to use new version of multisig contract that also include chain id
+  in the signature payload.
```
Comment (gromakovsky) [inline ChangeLog.md:18]: I would write explicitly that it's a sort of breaking change in sense that the new client won't work with previously deployed (old) multisig contracts.

File: package.yaml (hunk 3 above)
Comment (sras) [reply package.yaml:8]: Lol. Fixed.

File: ChangeLog.md (hunk 4 above)
Comment (sras) [reply ChangeLog.md]: Updated

**Comment by gromakovsky** [comment]
@sras please make a new release after merging this PR.

**Pull request merged**